### PR TITLE
[1/n] working upgrade and generic signer

### PR DESCRIPTION
**Pull request by VPhung24**
Huge work in progress. 

Cleaning up:

- migrate modules to their own packages
- generic signer on Auth

Plus,

- Lit: Social Auth, SIWE
- Turnkey: Passkey self custody 
- Wallet Connect, Metamask, 
- wrap specific packages for known dependency issues such as [lit dependency bug] `Module build failed: UnhandledSchemeError: Reading from "node:buffer" is not handled by plugins (Unhandled scheme).` fix by wrapping
- so so much more

---

<details open="true"><summary>Generated summary (powered by <a href="https://app.graphite.dev">Graphite</a>)</summary>

> ## TL;DR
> This pull request updates the `LitPasskey` class in the `src/auth/passkey/LitPasskey.ts` file and the `Snowball` class in the `src/snowball/Snowball.ts` file. It changes the `litNetwork` property in the `LitPasskey` class from "serrano" to "cayenne". It also modifies the `initSmartWallet` method in the `Snowball` class to handle the case when the authentication provider is `LitPasskey` and initializes the smart wallet accordingly. Additionally, it adds error handling for cases when the smart wallet is not initialized in various methods of the `Snowball` class.
> 
> ## What changed
> - In `src/auth/passkey/LitPasskey.ts`:
>   - The `litNetwork` property in the `LitPasskey` class is changed from "serrano" to "cayenne".
> 
> - In `src/snowball/Snowball.ts`:
>   - The `import` statements are updated to remove unused imports and add necessary imports.
>   - The `initSmartWallet` method is modified to handle the case when the authentication provider is `LitPasskey` and initializes the smart wallet accordingly.
>   - Error handling is added for cases when the smart wallet is not initialized in various methods of the `Snowball` class.
> 
> ## How to test
> 1. Checkout this branch.
> 2. Run the tests for the `LitPasskey` and `Snowball` classes.
> 3. Verify that the tests pass without any errors.
> 4. Manually test the functionality of the `LitPasskey` and `Snowball` classes to ensure that the changes work as expected.
> 
> ## Why make this change
> - The change in the `LitPasskey` class updates the `litNetwork` property to use the "cayenne" network instead of the "serrano" network. This change is made to align with the updated network configuration.
> - The modifications in the `Snowball` class are made to handle the case when the authentication provider is `LitPasskey` and initialize the smart wallet accordingly. This ensures that the smart wallet is properly initialized and can be used for further operations.
> - The error handling is added to handle cases when the smart wallet is not initialized in various methods of the `Snowball` class. This prevents potential errors and improves the reliability of the code.
</details>

### CODE REVIEW

File: src/auth/passkey/LitPasskey.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ export class LitPasskey extends Passkey {
 		) as WebAuthnProvider;
 
 		this.litNodeClient = new LitNodeClient({
-			litNetwork: "serrano",
+			litNetwork: "cayenne",
 			debug: true,
```
Comment (ericlewis) [inline src/auth/passkey/LitPasskey.ts:45]: should be configurable?

File: src/snowball/Snowball.ts [hunk 2]
```diff
@@ -93,7 +122,11 @@ export class Snowball {
...
 	async authenticate(): Promise<void> {
 		try {
-			return await this.auth.authenticate();
+			await this.auth.authenticate();
+
+			this.smartWallet = await this.initSmartWallet();
+
+			return Promise.resolve();
 		} catch (error) {
 			return Promise.reject(`authenticate failed ${error}`);
```
Comment (ericlewis) [inline src/snowball/Snowball.ts:131]: you don't need the resolve / reject here, you can rethrow error from the catch or simply don't try/catch for the same result (with the exception that you aren't rewriting the result)

File: src/snowball/Snowball.ts [hunk 3]
```diff
@@ -146,6 +190,9 @@ export class Snowball {
     hash: string;
   }> {
 		try {
+			if (!this.smartWallet) {
```
Comment (ericlewis) [inline src/snowball/Snowball.ts:193]: this check is appear a lot, maybe consider turning it into a function and using it that way.

**Review by ericlewis — CHANGES_REQUESTED** [review]
looks good. a few comments.

File: src/snowball/Snowball.ts (hunk 3 above)
Comment (VPhung24) [reply src/snowball/Snowball.ts:193]: will do! meant to be a patch before the major rewrite (starting with the next pull request making each module its own package)

**Review by ericlewis — APPROVED** [review]

**Comment by VPhung24** [comment]
### Merge activity

* **Feb 5, 4:17 PM EST**: @VPhung24 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/snowball-tools/snowball-ts-sdk/17).

**Pull request merged**